compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/semantics/analyzer/Types.java

```diff
@@ -2736,25 +2781,16 @@ BType getTypeForFiniteTypeValuesAssignableToType(BFiniteType finiteType, BType t
      * @return           a single type or a new union type if at least one member type of the union type is
      *                      assignable to targetType, else semanticError
      */
-    BType getTypeForUnionTypeMembersAssignableToType(BUnionType unionType, BType targetType) {
+    BType getTypeForUnionTypeMembersAssignableToType(BUnionType unionType, BType targetType, SymbolEnv env) {
         List<BType> intersection = new LinkedList<>();
 
         // type FooOne "foo"|1;
         // type FooBar "foo"|"bar";
         // unionType - boolean|FooOne, targetType - boolean|FooBar
         unionType.getMemberTypes().forEach(memType -> {
```

Comment:
Can you check if we can replace this entire method with a direct call to `getTypeIntersection` instead?

Reply:
I did not get what you mean here, As I understood it is to replace `getTypeForUnionTypeMembersAssignableToType` -> `getTypeIntersection` and remove the `getTypeForUnionTypeMembersAssignableToType` from the switch-case check in  getTypeIntersection. Is that correct?

Reply:
I missed that this was getting called from `getTypeIntersection`. This is OK.